en/apps/module_template.md

```diff
@@ -1,10 +1,24 @@
 # Module Template for Full Applications
 
-The PX4 Firmware contains a template for writing a new application (module) that runs as a [task](../concept/architecture.md#runtime-environment) on its own stack:
-[src/templates/module](https://github.com/PX4/Firmware/tree/master/src/templates/module).
+An application can be written to run as either a task (a module with its own stack and process priority) or as a work queue item (multiple modules share the same stack).
+In most cases a work queue item can be used, as this minimizes resource usage.
+
+> **Note** [Architectural Overview > Runtime Environment](../concept/architecture.md#runtime-environment) provides more information about stacks and work queues.
 
+<span></span>
 > **Note** All the things learned in the [First Application Tutorial](../apps/hello_sky.md) are relevant for writing a full application.
 
+
+## Work Queue Item
+
+The PX4 Firmware contains a template for writing a new application (module) that runs as a work queue item: 
+[src/examples/work_item](https://github.com/PX4/Firmware/tree/master/src/examples/work_item).
+
```

Comment:
It is not clear to me what it is that you do that makes the module behave as a work queue or as a task. We can save users a lot of time by spelling this out.

E.g. Just enough for people not to have to guess what is relevant.
1. In Tasks section we should explain what makes a module - derives from X, Y, Z. uses CMAKE X, Y, Z.
2. Here explain the delta - again , implements interface X, does Y. 

By inspection I THINK that for a work queue item;
1. CMAKE
   ```
	DEPENDS
		px4_work_queue
   ```
1. Includes https://github.com/PX4/Firmware/blob/master/platforms/common/include/px4_platform_common/px4_work_queue/ScheduledWorkItem.hpp
2. Derives from `ScheduledWorkItem`
3. Implements the `ScheduledWorkItem::Run()` method and sets the scheduling rate

It also looks like you might have to do something in the ModuleBase::taskSpawn to actually create it on the work queue?

Can you clarify the essentials and essential differences?

Reply:
Yes this is all correct. In spawning the tasks you say it's a work queue item: https://github.com/PX4/Firmware/blob/master/src/examples/work_item/WorkItemExample.cpp#L100
Then it schedules itself in `init`: https://github.com/PX4/Firmware/blob/master/src/examples/work_item/WorkItemExample.cpp#L54.